realm/realm-annotations-processor/src/main/java/io/realm/processor/RealmProxyClassGenerator.java

```diff
@@ -1553,6 +1553,10 @@ private void emitCreateOrUpdateUsingJsonObject(JavaWriter writer) throws IOExcep
         writer.emitEmptyLine();
     }
 
+    // FIXME: Since we need to check the PK in stream before create an object, this is now using copyToRealm instead of
```

Comment:
That is only for `createOrUpdateAllFromJson`, for updating. by creating directly from stream, we set the properties one by one. If this is not acceptable, I would like to implement the #777 first since it doesn't seem to be too difficult.